returnn/hdf.py

```diff
@@ -207,18 +208,32 @@ class BlissToPcmHDFJob(Job):
     compatible with the RETURNN HDFDataset
     """
 
+    class BaseStrategy:
+        pass
+
+    @dataclass(frozen=True)
+    class PickNth(BaseStrategy):
+        channel: int
+
+    __sis_hash_exclude__ = {"multi_channel_strategy": "none"}
```

Comment:
this needs to be changed as well. But would this work, or would a different BaseStrategy instance be a problem?
```suggestion
    __sis_hash_exclude__ = {"multi_channel_strategy": BaseStrategy()}
```

Reply:
I request changes on my own pr.. 🤷 
@JackTemaki can we revert this merge, as this job is used in production but the sis hash exclude does not work I.e. hashes will be different.

Reply:
I think this would not work:

```python
class Demo():
    pass
    
exclude = {"x": Demo()}

def f(x = Demo()):
    if x != exclude["x"]:
        print("would include")
    else:
        print("would not include")
 ```
 yields "would include", and this is the check sisyphus does.

Reply:
So the same example with:
```python
class Demo():
    def __eq__(self, other):
        return type(other) == Demo
        ```
works.